### PR TITLE
Better handling of original timestamp

### PR DESCRIPTION
**Pull request by ph1lm**
*Description of changes:*
Predicates are added for point-in-time replication.
Also, an opportunity to replicate the original timestamp was added.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ph1lm** [comment]
Hi @nwheeler81 
Sorry for delay 
I took some time to get it approved

File: glue/README.MD [hunk 1]
```diff
@@ -136,6 +136,20 @@ on the target table before running the following command
...
+the Client-Side Timestamps feature should be [enabled](https://docs.aws.amazon.com/keyspaces/latest/devguide/client-side-timestamps.html) 
+on the target table before running the following command
+```shell
+   cqlreplicator --state run --tiles 8 --landing-zone s3://cql-replicator-1234567890-us-west-2 --region us-west-2 \
+                 --writetime-column col3 --src-keyspace ks_test_cql_replicator --src-table test_cql_replicator \ 
+                 --trg-keyspace ks_test_cql_replicator --trg-table test_cql_replicator --json-mapping '
+                {
+                 "replication": {
+                   "replicateWithTimestamp": true
+                }'
```
Comment (nwheeler81) [inline glue/README.MD:150]: missing `}`

**Comment by nwheeler81** [comment]
Tested use-cases:
Table structure (id text PRIMARY KEY, code timeuuid, value text):  
1. ```./cqlreplicator --state run --tiles 1 --landing-zone "s3://cql-replicator-account-us-east-1-pitr" --region us-east-1 --worker-type G.025X --writetime-column value --src-keyspace ks_pitr --src-table tbl_pitr --trg-keyspace ks_pitr --trg-table tbl_pitr --env issue151 --json-mapping '{ "replication": {"replicateWithTimestamp": true}}' ``` - **passed** 

2. ```./cqlreplicator --state run --tiles 1 --landing-zone "s3://cql-replicator-account-us-east-1-pitr" --region us-east-1 --worker-type G.025X --srf 1722270672200697 --writetime-column value --src-keyspace ks_pitr --src-table tbl_pitr --trg-keyspace ks_pitr --trg-table tbl_pitr --env issue151 --json-mapping ' {"replication": { "replicateWithTimestamp": true,"pointInTimeReplicationConfig": {"predicateOp": "lessThan"} }}'``` - **passed**

3. ```./cqlreplicator --state run --tiles 1 --landing-zone "s3://cql-replicator-account-us-east-1-pitr" --region us-east-1 --worker-type G.025X --srf 1722270672200697 --writetime-column value --src-keyspace ks_pitr --src-table tbl_pitr --trg-keyspace ks_pitr --trg-table tbl_pitr --env issue151 --json-mapping ' {"replication": { "replicateWithTimestamp": true,"pointInTimeReplicationConfig": {"predicateOp": "greaterThan"} }}'``` - **passed**

**Comment by nwheeler81** [comment]
@michaelraney could you please review #154

**Comment by michaelraney** [comment]
Interested to know what is the use case for "lessThanOrEqual"

**Comment by ph1lm** [comment]
@michaelraney We used it for our migration from C* to K*. 

We implemented dual-write capability in our application and enabled it in prod so it was writing to both C* and K*, but still reading from C*, and we captured a timestamp. So, all new data after this timestamp is replicated from C* to K* by dual-write.

After that, we launched the cql-replicator with lessThanOrEqual=timestamp option.
So, all old data before the timestamp is replicated from C* to K* by cql-replicator.

After cql-replicator finished, we disabled dual-write and switched references from C* to K* for both reads and writes.

File: glue/README.MD (hunk 1 above)
Comment (ph1lm) [reply glue/README.MD:151]: Fixed

**Comment by michaelraney** [comment]
Great thank you,

**Review by michaelraney — APPROVED** [review]
Looks good

**Pull request merged**